flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImpl.java

```diff
@@ -157,6 +151,14 @@ public void addInputData(
                 checkpointId,
                 info,
                 startSeqNum);
+        if (isCheckpointSubsumedOrAborted(checkpointId)) {
+            LOG.debug(
+                    "The checkpoint {} of task {} has been aborted, so don't addInputData.",
+                    checkpointId,
+                    taskName);
+            closeBuffers(iterator);
+            return;
+        }
         enqueue(write(checkpointId, info, iterator), false);
```

Comment:
What will happen if there is a race condition between this `addInputData` call from the netty threads, and for example `abort` call?

1. (Netty thread) calling `addInputData`
2. (Netty thread) checking `isCheckpointSubsumedOrAborted`, method returns false
3. (Netty thread) freezes for a 1ms
4. (Task thread) calls `abort()`
5. (Netty thread) wakes up and finishes `addInputData` method by calling `enqueue(write(checkpointId, info, iterator), false);`

Will this lead to a resource leak? If not, then why do we need `if (isCheckpointSubsumedOrAborted(checkpointId))` check here in the first place? `isCheckpointSubsumedOrAborted` is quite costly due to lock acquisition after all.

Reply:
~this issue is no longer relevant~ we decided to go with another approach